File: pythonProject/proj/Animal/Turtle.py
from proj.Animal.myAnimal import *

class Turtle(Animal):
    def __init__(self, world: World, point: Point, initiative, power, age):
        super().__init__(world, point, 2, 2, 0)
        self._power = power
        self.age = age
        self._typ = 3
        self._sign = 'T'
    def getColor(self):
        return (31, 122, 31)

    def Akcja(self):
        
        left = True
    
        right = True
    
        up = True
    
        down = True
   
        if (0 + self.getPosition().getX() - 1 < 0) :
            left = False
    
        elif (0 + self.getPosition().getX() + 2 > 0 + self._world.getX()) :
            right = False
    
        if (self.getPosition().getY() - 1 < 0) :
            up = False
    
        elif (self.getPosition().getY()+ 2 > 0 + self._world.getY()) :
            down = False
    

        ifGo = randint(0, 100)
        if (ifGo > 75):
            while (True) :

                a = randint(0, 100)

                if (a >= 0 and a < 25 and up == True) :
                    self._point.changePositionToUp()
                    break
    
                elif (a >= 25 and a < 50 and down == True) :
                    self._point.changePositionToDown()
                    break



                elif (a >= 50 and a < 75 and left == True) :
                    self._point.changePositionToLeft()
                    break

                elif (a >= 75 and right == True) :
                    self._point.changePositionToRight()
                    break



            if (self._world.whoIs(self.getPosition(), self) != None and self._world.whoIs(self.getPosition(), self).getSign() != self.getSign()) :
                self._world.whoIs(self.getPosition(), self).Kolizja(self)

            elif (self._world.whoIs(self.getPosition(), self) != None and self._world.whoIs(self.getPosition(), self).getSign() == self.getSign() and self.getAge() > 10 and self._world.whoIs(self.getPosition(), self).getAge() > 10):
                self.breeding(self._world.whoIs(self._point, self))
        self.plusAge()

    def Kolizja(self, org):
        if (org.getSila() > 5) :
            self._world.deleteOrganizm(self)



        else :

            org.getPosition().changePositionToLeft()

File: javaProject/src/main/java/Animals/Turtle.java
package Animals;

import project_java.Animal;
import project_java.Organizm;
import project_java.Point;
import project_java.World;

import javax.imageio.ImageIO;
import java.awt.*;
import java.io.IOException;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.Random;

public class Turtle extends Animal {
    public Turtle(World world, Point position){
        super(world,position);
        age = 0;
        power = 2;
        inicjatywa = 1;
        sign = 't';
        name = "turtle";
        typ = 5;
        isAnimal = true;
    }
    public Turtle(World world,Point position, int age, int power, int init){
        super(world,position,age,power,init);
        isAnimal = true;
        sign = 't';
        name = "turtle";
        typ = 5;

        //oldPosition = new Point(0,0);
    }
    public Color getColor(){
        return new Color(96, 85, 10);
    }

    @Override
    public void Akcja() {
        boolean left = true;
        boolean right = true;
        boolean up = true;
        boolean down = true;
     oldPosition = position;
        if (0+this.getPosition().getX() - 1 < 0) {
            left = false;
        }
        else if (0 + this.getPosition().getX() + 2> 0 + world.getX()) {
            right = false;
        }
        if (this.getPosition().getY() - 1 < 0) {
            up = false;
        }
        else if (this.getPosition().getY()+ 2> 0 + world.getY()) {
            down = false;
        }

        Random random = new Random();
        int ifGo = random.nextInt(100);
        if(ifGo> 75){
        while (true) {



            int a = random.nextInt(100);

            if (a >= 0 && a < 25 && up == true) {
                position.changePositionUp();

                break;
            }
            else if (a >= 25 && a < 50 && down == true) {
                position.changePositionDown();
                break;
            }

            else if (a >= 50 && a < 75 && left == true) {
                position.changePositionLeft();
                break;
            }
            else if (a >= 75 && right == true) {
                position.changePositionRight();
                break;
            }

        }
        if (world.whoIs(this.getPosition(), this) != null && world.whoIs(this.getPosition(), this).getSign() != this.getSign()) {
            world.whoIs(this.getPosition(), this).Kolizja(this);
        }
        else if(world.whoIs(this.getPosition(), this) != null && world.whoIs(this.getPosition(), this).getSign() == this.getSign() && this.getAge() > 10 && world.whoIs(this.getPosition(), this).getAge()> 10){
            breeding(world.whoIs(this.position, this));
            this.setPosition(oldPosition);

        }}



        plusAge();
    }
    public void Kolizja(Organizm org){
        if (org.getSila() > 5) {
            world.deleteOrganizm(this);
            world.addInformation(org.getName()+" killed "+this.getName() + " on position "+ this.position.getStringPosition() +"\n");

        }
        else {
            //org.getPosition().setPositoion(org.goldPosition);
            org.getPosition().changePositionLeft();
         // org.setPosition(org.getOldPosition());
            world.addInformation(this.getName()+" kicked out  "+org.getName() + " from position "+ this.position.getStringPosition() +"\n");
        }
    }
    public String getName(){
        return "Turtle";
    }
    public Image getImage()  {

        // String fileName = "file:/Users/bitesaitzzz/IdeaProjects/demo/src/main/java/Animals/images/man.png";
        URL url = null;
        try {
            url = new URL("file:" + System.getProperty("user.dir") + "/src/main/java/images/turtle.png");

        } catch (MalformedURLException e) {
            throw new RuntimeException(e);
        }
        if (url == null) {
            throw new RuntimeException("Resource not found: ");
        }
        try {
            return ImageIO.read(url);
        } catch (IOException e) {
            throw new RuntimeException("Failed to load image: ", e);
        }
    }
}
